network/address_test.go

```diff
@@ -537,6 +537,73 @@ func (s *AddressSuite) TestSelectInternalMachineAddress(c *gc.C) {
 	}
 }
 
+type selectInternalAddressesTest struct {
+	about           string
+	addresses       []network.Address
+	machineLocal    bool
+	expectedIndices []int
+	expectedOk      bool
+}
+
+// expected returns the expected address for the test.
+func (t selectInternalAddressesTest) expected() ([]network.Address, bool) {
+	if t.expectedIndices == nil {
+		return nil, t.expectedOk
+	}
+	expected := make([]network.Address, len(t.expectedIndices))
+	for i, idx := range t.expectedIndices {
+		expected[i] = t.addresses[idx]
+	}
+	return expected, t.expectedOk
+}
+
+var selectInternalAddressesTests = []selectInternalAddressesTest{
+	{
+		"machine-local addresses are selected when machineLocal is true",
+		[]network.Address{
+			network.NewScopedAddress("127.0.0.1", network.ScopeMachineLocal),
+			network.NewScopedAddress("cloud-local.internal", network.ScopeCloudLocal),
+			network.NewScopedAddress("fc00::1", network.ScopePublic),
+		},
+		true,
+		[]int{0},
```

Comment:
any chance we could change this to something like []string{address1, address2}
it is much easier to reason about the test seeing the raw strings, rather than the integer offsets in some other slice.